pennylane/data/attributes/array.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains AttributeType definition for numpy arrays."""
+
+import numpy
+from numpy.typing import ArrayLike
```

Comment:
Is `pennylane.numpy.tensor` basically used everywhere in pennylane instead of `numpy.array`? If both are used I'd prefer to define a separate type for `tensor`?

Reply:
not exactly. People use the ML framework of their choosing from what we support, and those are autograd (wrapped behind `pennylane.numpy.tensor`), jax, tensorflow or torch. You can use vanilla numpy as well, but there's no auto-differentiation support in that case. `qml.TensorLike` will just be a superset of `numpy.typing.ArrayLike`, and cover all values that PennyLane can support. It so happens that today's datasets all use autograd tensors, but one day this can (dare I saw will!) change